extract_metadata.py

```diff
@@ -0,0 +1,111 @@
+from dandi.dandiapi import DandiAPIClient
+import pandas as pd
+from pathlib import Path
+
+modalities = {'oct': 'PS-OCT',
+              'df': 'Dark field microscopy',
+              'xpct': 'HiP-CT',
+              'dwi': 'Diffusion MRI',
+              'fluo': 'Light sheet microscopy',
+              'photo': 'Blockface photo'}
+
+# Create dataframe of all assets across all datasets
+def extract_assets():
+    client = DandiAPIClient("https://api.lincbrain.org/api")
+    client.dandi_authenticate()
+    dandisets = client.get_dandisets()
+
+    print(f"Processing {sum(1 for _ in dandisets)} Datasets on lincbrain.org")
+
+    df = pd.DataFrame(columns=["Dataset",
+                            "Version",
+                            "Subject", 
+                            "Modality",
+                            "Path",
+                            "Filename",
+                            "Extension",
+                            "Directory", # Top-level directory (e.g. source data, raw data, derivatives)
+                            'Size (bytes)'])
+
+    for dataset in dandisets:
+        latest_dataset = dataset.for_version('draft')
+
+        # Exclude datasets '000048' (OpenBNB) and '000004' (Mouse LSM)
+        if latest_dataset.identifier not in ['000048', '000004']:
+            for asset in latest_dataset.get_assets():
+                asset_split = asset.path.split('/')
+
+                print(f"Dataset: {latest_dataset}; Asset: {asset_split[-1]:<80}", end='\r')
+
+                metadata = asset.get_metadata()
+                metadata_dict = metadata.model_dump(mode='json', exclude_none=True)
+
+                subject = 'Unknown'
+                for part in asset_split:
+                    if part.startswith("sub-"):
+                        subject = part.split("sub-")[1].split('_')[0]
+                        break
+
+                if subject == 'Unknown' and any(filename in asset_split[-1].lower() for filename in ['dataset_description.json', 'participants.tsv', 'readme.md', 'samples.tsv']):
+                    subject = 'n/a'
+
+                modality = next((value for key, value in modalities.items() 
+                                if key in asset_split[-1].lower()), 
+                                'Unknown')
+
+                suffix = Path(asset.path).suffixes[0][1:] if Path(asset.path).suffixes else ''
+
+                df.loc[len(df)] = [latest_dataset.identifier,
```

Comment:
I think this operation might be slightly slower than creating a list of lists, and then appending to the dataframe at the end of the loop -- e.g.:

```
lastest_datasets = []
for dataset in datasets:
     -fun logic here-
     lastest_datasets.append({
            "Dataset": latest_dataset.identifier,
            "Version": latest_dataset.version.identifier,
            "Subject": subject,
            "Modality": modality,
            "Path": asset.path,
            "Filename": asset_split[-1],
            "Extension": suffix,
            "Directory": asset_split[0],
            "Size (bytes)": metadata_dict.get("contentSize", 0)
        })
        
return pd.DataFrame(lastest_datasets)
```

Reply:
Great idea.  I just pushed an update and tested that it works.